internal/metrics/metrics.go

```diff
@@ -0,0 +1,57 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/RHEnVision/provisioning-backend/internal/kafka"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/version"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	metricNameSourceAvailabilityCheckRequestTotal    = "source_availability_check_request_total"
+	metricNameSourceAvailabilityCheckRequestDuration = "source_availability_check_request_duration_ms"
+)
+
+var TotalAvailabilityCheckReqs = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name:        metricNameSourceAvailabilityCheckRequestTotal,
+		Help:        "availability Check requests count partitioned by type (aws/gcp/azure), source status, component and error",
+		ConstLabels: prometheus.Labels{"service": version.PrometheusLabelName},
+	},
+	[]string{"type", "status", "component", "error"},
+)
+
+var AvailabilityCheckReqsDuration = prometheus.NewHistogramVec(
+	prometheus.HistogramOpts{
+		Name:        metricNameSourceAvailabilityCheckRequestDuration,
+		Help:        "Availability check Request duration partitioned by type and error",
+		ConstLabels: prometheus.Labels{"service": version.PrometheusLabelName},
+	},
+	[]string{"type", "error"},
+)
+
+func ObserveAvailablilityCheckReqsDuration(provider models.ProviderType, start time.Time, err error) {
+	errString := ""
+	if err != nil {
+		errString = err.Error()
+	}
+	AvailabilityCheckReqsDuration.WithLabelValues(provider.String(), errString).Observe(float64(time.Since(start).Nanoseconds()) / 1000000)
+}
+
+func IncTotalAvailablilityCheckReqs(provider models.ProviderType, component string, StatusType kafka.StatusType, err error) {
+	errString := ""
+	if err != nil {
+		errString = err.Error()
+	}
+	TotalAvailabilityCheckReqs.WithLabelValues(provider.String(), string(StatusType), component, errString).Inc()
+}
+
+func RegisterTotalAvailablilityCheckReqs() {
+	prometheus.MustRegister(TotalAvailabilityCheckReqs)
+}
+
+func RegisterAvailabilityCheckReqsDuration() {
+	prometheus.MustRegister(AvailabilityCheckReqsDuration)
+}
```

Comment:
There could be simply `RegisterAll` or whatever, can be refactored later this is fine.